neurons/validator.py

```diff
@@ -478,27 +169,34 @@ def run(self):
                     f"Validator running on step({self.step}) block({self.block})."
                 )
 
-                # Run multiple forwards concurrently.
+                # Ensure miner or validator hotkey is still registered on the network.
+                self.check_registered()
+
+                # Evaluate the next batch of miners.
                 next_batch_delay_secs = self.loop.run_until_complete(
-                    self.run_next_eval_batch()
+                    self.evaluator.run_next_eval_batch()
                 )
+                self._on_eval_batch_complete()
 
-                next_batch_start_time = datetime.datetime.utcnow() + datetime.timedelta(
+                # Set the next batch start time.
+                next_batch_start_time = dt.datetime.utcnow() + dt.timedelta(
```

Comment:
I realize this is from before but kind of odd to add now here just to subtract it on 198. Should we just wait the seconds down there?

Reply:
Agreed it looks weird. We do this because we set the weights in-between which, I assume, can take a significant amount of time.

Reply:
Ah, fair.